chill-algebird/src/main/scala/com/twitter/chill/algebird/AlgebirdSerializers.scala

```diff
@@ -95,3 +86,42 @@ class HLLMonoidSerializer extends KSerializer[HyperLogLogMonoid] {
     hllMonoids.getOrElseUpdate(bits, new HyperLogLogMonoid(bits))
   }
 }
+
+class SketchMapSerializer[K, V](implicit skmMonoid: SketchMapMonoid[K, V], valueMonoid: Monoid[V])
+  extends KSerializer[SketchMap[K, V]] {
+
+  def write(kryo: Kryo, output: Output, skm: SketchMap[K, V]) {
+    val rows = skm.valuesTable.rows
+    val cols = skm.valuesTable.cols
+
+    val values: IndexedSeq[(Int, Int, V)] =
+      for (
+        r: Int <- 0 to rows - 1; c: Int <- 0 to cols - 1 if skm.valuesTable.getValue((r, c)) != valueMonoid.zero
+      ) yield (r, c, skm.valuesTable.getValue((r, c)))
+
+    output.writeInt(rows, true)
+    output.writeInt(cols, true)
+    kryo.writeClassAndObject(output, values.toList)
```

Comment:
is the toList needed there? That is going to be a full iteration through things again.

A few notes:

1. you could use `0 until rows`. rather than the `-1`.
2. map/flatMap on Range is not lazy, so this is going to fully materialize the filtered table twice, once in the for expression, once in .toList.
3. prefer valueMonoid.isNonZero rather than the equality check (it's janky in the end, and I wish we had a good Equivalence typeclass, but its what exists now).